src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -57,7 +57,9 @@
     private final DanglingIndicesState danglingIndicesState;
 
     @Nullable
-    private volatile MetaData currentMetaData;
+    private volatile MetaData previousMetaData;
+
+    private final Set<String> previouslyWrittenIndices = Collections.synchronizedSet(new HashSet<String>());
```

Comment:
can we change this to volatile and use an ImmutableSet when we assign it (the members shouldn't be changed)